src/foam/core/AbstractFObjectPropertyInfo.java

```diff
@@ -105,7 +105,7 @@ public boolean hardDiff(FObject o1, FObject o2, FObject diff) {
      *  X.get("DAO").put(obj);
      * In this case: the before value inside the model is loss, so can not find difference
      */
-    if ( (this.get(o1) == this.get(o2)) ) {
+    if ( ( this.get(o1) == null && this.get(o2) != null ) || ( this.get(o1) == this.get(o2) ) ) {
```

Comment:
I've changed this to the following code which is more general because it also checks the case where o1 != null and o2 does. Have merged and will close this PR.

    Object fo1 = this.get(o1), fo2 = this.get(o2);
    if ( fo1 == fo2 || ( ( fo1 == null ) != ( fo2 == null ) ) ) {